posts/2021-11-03-tensorflow-gpu.rst

```diff
@@ -41,8 +41,15 @@ You can now select between GPU enabled (default) and CPU packages using the `ten
     # OR
     conda install tensorflow-gpu -c conda-forge
 
-When installing the `tensorflow` package, the package resolution will now default to the `tensorflow-gpu` package (that will also work on CPU). If you want the slimmer "cpu-only" package, then you can install `tensorflow-cpu` directly.
+When installing the `tensorflow` package, the package resolution will now default to the `tensorflow-gpu` package if the local machine has a GPU. Note that GPU-enabled packages can also work on CPUs, but one would need to override the enviornment variable `CONDA_CUDA_OVERRIDE` like below. This could be handy if you are in a situation where your current node (e.g. login node) on an HPC does not have GPUs, but the compute nodes with GPUs do not have interenet access.
 
+.. code-block:: bash
+
+    CONDA_CUDA_OVERRIDE="11.2" conda install tensorflow -c conda-forge
+    # OR
+    CONDA_CUDA_OVERRIDE="11.2" mamba install tensorflow -c conda-forge
+
+Note that you should select the cudatoolkit version most appropraite for your GPU; currently, we have "10.2", "11.0", "11.1", and "11.2" available. If you want the slimmer "cpu-only" package, then you can install `tensorflow-cpu` directly. At the time of writing (February 2022), on a machine without a GPU, one would always get the `-cpu` variant unless overriden like above. This decision has been made to allow greater accessibility for users with limited bandwidth and resources.
```

Comment:
This CUDA selection is done automatically when you have a GPU by calling `nvidia-smi`.

Reply:
Let me try to incorporate this into the text, paging @jakirkham for help in how to best phrase this.